samza-log4j2/src/main/java/org/apache/samza/logging/log4j2/StreamAppender.java

```diff
@@ -67,48 +69,52 @@
 import org.apache.samza.system.SystemStream;
 import org.apache.samza.util.ExponentialSleepStrategy;
 import org.apache.samza.util.HttpUtil;
+import org.apache.samza.util.MetricsReporterLoader;
 import org.apache.samza.util.ReflectionUtil;
 
 @Plugin(name = "Stream", category = "Core", elementType = "appender", printObject = true)
 public class StreamAppender extends AbstractAppender {
 
-  private static final String JAVA_OPTS_CONTAINER_NAME = "samza.container.name";
-  private static final String JOB_COORDINATOR_TAG = "samza-job-coordinator";
-  private static final String SOURCE = "log4j-log";
+  protected static final String JAVA_OPTS_CONTAINER_NAME = "samza.container.name";
+  protected static final String JOB_COORDINATOR_TAG = "samza-job-coordinator";
+  protected static final String SOURCE = "log4j-log";
```

Comment:
In my use case (extend existing StreamAppender), I do use small pieces of the changed variables. Just in case in the future, we want to expose more objects or just make extend-friendly feature thoroughly, I prefer to modify all necessary variables (seem useful for child classes) to protected instead of only changing the variables that I intend to use.

Reply:
within Samza i see classes which have only some of their variables exposed as protected and keep the rest private. So i think it is an acceptable pattern to have partial protected. We can expose more as the need arises.

So the current need to extend is coming from having a usecase to support another transformation right .. this PR can "make the parts extendable for different transformation/serde" and that i believe will be acceptable too.

Reply:
@bkonold Do you have ideas about this issue?

Reply:
Agree I've seen partial protection in several places of Samza so I think it would be fine here. Labeling only what you need to as protected is also better for the reader because it will restrict the set of functionality a subclass can override. IMO, I'd be in favor of labeling only what you need to as protected.